scripts/node_api_check.py

```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/python3
+
+import requests as r
+import argparse
+import sys
+import pandas as pd
+import json
+
+class MainFunctions:
+
+    def __init__(self):
+        self.api_url = "https://validator.nymtech.net/api/v1"
+        #self.api_existing_endpoints_url = "https://validator.nymtech.net/api/v1/openapi.json"
+        self.api_endpoints_json = "api_endpoints.json"
+
+    def display_results(self,args):
+        id_key = args.id
+        gateways_unfiltered, mixnodes_unfiltered = self.get_unfiltered_data()
+        gateways_df = self._json_to_dataframe(gateways_unfiltered)
+        mixnodes_df = self._json_to_dataframe(mixnodes_unfiltered)
+        mode, node_series = self.node_type_check(id_key, gateways_df, mixnodes_df)
+        print(f"Node type = {mode}")
+        api_data, swagger_data = self.get_node_data(mode, node_series, id_key)
+        api_data = pd.Series(api_data)
+        swagger_data = pd.Series(swagger_data)
+        print(node_series.T, "\n")
+        print(api_data, "\n")
+        print(swagger_data)
+
+    def node_type_check(self,id_key, gateways_df, mixnodes_df):
+        if id_key in mixnodes_df['mixnode_details.bond_information.mix_node.identity_key'].values:
+
+            node = mixnodes_df.loc[mixnodes_df['mixnode_details.bond_information.mix_node.identity_key'] == id_key]
+            mode = "mixnode"
+        elif id_key in gateways_df['gateway_bond.gateway.identity_key'].values:
+            node = gateways_df.loc[gateways_df['gateway_bond.gateway.identity_key'] == id_key]
+            mode = "gateway"
+        else:
+            print(f"The identity key '{id_key}' does not exist.")
+
+        return mode, node
+
+    def get_unfiltered_data(self):
+        gateways_unfiltered = r.get(f"{self.api_url}/status/gateways/detailed-unfiltered").json()
+        mixnodes_unfiltered = r.get(f"{self.api_url}/status/mixnodes/detailed-unfiltered").json()
+        return gateways_unfiltered, mixnodes_unfiltered
+
+    def get_mixnode_data(self, node_series, id_key):
+        mix_id = int(node_series["mixnode_details.bond_information.mix_id"])
+
+
+    def get_node_data(self,mode, node_series, id_key):
+        #endpoint_json = self.get_api_endpoints()
+        identity = id_key
+        endpoint_json = self.api_endpoints_json
+        with open(endpoint_json, "r") as f:
+            dicts = json.load(f)
+            enpoints = dicts[mode]
+            swagger = dicts["swagger"]
+        api_data = {}
+        swagger_data = {}
+        if mode == "gateway":
+            host = str(node_series["gateway_bond.gateway.host"])
+            api_data["API ENPOINTS"] = "RESPONSE"
+            for key in enpoints:
+                endpoint = key.replace("{identity}", identity)
+                url = f"{self.api_url}{endpoint}"
+                value = r.get(url)
+                api_data[endpoint] = value
+            swagger_data["SWAGGER ENDPOINTS"] = "RESPONSE"
+            for key in swagger:
```

Comment:
just to make sure, you're iterating over nym-node endpoints here, right? and **not** nym-api?

Reply:
I iterate over validator.nymtech.net/api/v1 and over swagger endpoints for gws as below. Should I add some? My plan is to eventually dump api_json_file and use the online openapi.json as a source.